n_hooks.c

```diff
@@ -821,16 +831,57 @@ bool NoteHardReset()
 /*!
   @brief  Perform a JSON request to the Notecard using the currently-set
   platform hook.
-  @param   json the JSON request.
-  @param   jsonResponse (out) A buffer with the JSON response.
+  @param   request the JSON request.
+  @param   response (out) A buffer with the JSON response.
   @returns NULL if successful, or an error string if the transaction failed
   or the hook has not been set.
 */
 /**************************************************************************/
-const char *NoteJSONTransaction(char *json, char **jsonResponse)
+const char *NoteJSONTransaction(char *request, char **response)
 {
     if (notecardTransaction == NULL || hookActiveInterface == interfaceNone) {
         return "i2c or serial interface must be selected";
     }
-    return notecardTransaction(json, jsonResponse);
+    return notecardTransaction(request, response);
+}
+
+/**************************************************************************/
+/*!
+  @brief  Receive bytes over from the Notecard using the currently-set
+  platform hook.
+  @param   buffer A buffer to receive bytes into.
+  @param   size (in/out)
+            - (in) The size of the buffer in bytes.
+            - (out) The length of the received data in bytes.
+  @param   delay Respect delay standard transmission delays.
+  @param   timeoutMs The maximum amount of time, in milliseconds, to wait for
+            serial data to arrive. Passing zero (0) disables the timeout.
+  @param   overflow (out) The contents did not fit inside the provided buffer.
```

Comment:
I would be more specific. If it's just what the comment here says it is, why isn't overflow a bool?